std/path.d

```diff
@@ -1280,6 +1280,11 @@ if (isSomeChar!C1 && isSomeChar!C2)
     assert(withExtension("file.ext"w.byWchar, ".").array == "file."w);
 }
 
+@safe unittest
+{
+    assert(chainPath("directory", "file").withExtension(".ext").array == "directory/file.ext");
```

Comment:
```suggestion
    assert(chainPath("directory", "file").withExtension(".ext").array == "directory".buildPath("file.ext");
```

Reply:
Thank you! At a moment of bliss, I forgot that other operating systems exist.